rpc/src/lib.rs

```diff
@@ -17,6 +17,28 @@
 //! Parity RPC.
 
 #![warn(missing_docs, unused_extern_crates)]
+#![cfg_attr(feature = "cargo-clippy", warn(clippy::all, clippy::pedantic))]
+#![cfg_attr(
+	feature = "cargo-clippy",
+	allow(
+		clippy::empty_enum,
+		clippy::redundant_field_names,
+		// things are often more readable this way
+		clippy::cast_lossless,
+		clippy::module_name_repetitions,
+		clippy::single_match_else,
+		clippy::type_complexity,
+		clippy::use_self,
+		clippy::zero_prefixed_literal,
```

Comment:
Do we actually have zero prefixed literals?

Reply:
No, remove? 

Well, this a first attempt to get some consensus around lints to allow/warn/deny in the code base. To make it easier to use the same in all the crates